assets/components/button/_sharedButton.jsx

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React, { type Node } from 'react';
+
+import SvgArrowRightStraight from 'components/svgs/arrowRightStraight';
+import { classNameWithModifiers } from 'helpers/utilities';
+
+import './button.scss';
+
+// ----- PropTypes ----- //
+
+type Appearance = 'primary' | 'green' | 'greenHollow' | 'greyHollow';
```

Comment:
we are missing, like, _a lot_ of appearances, I'd suggest to add them as we go/need rather than all at once?

Reply:
What about separating out the hollow/solid from the color, as @zeftilldeath did in his examples on https://interactive.guim.co.uk/embed/article-embeds/HiZef/embed.html?

```html
<button class="button button--large button--solid button--lozenge button--color-yellow" type="button">
  Submit
</button>
```

```html
<button class="button button--large button--hollow button--lozenge button--color-neutral" type="button">
  No, click Me!
</button>
```